include/oneapi/dpl/internal/histogram_extension_defs.h

```diff
@@ -0,0 +1,48 @@
+// -*- C++ -*-
```

Comment:
Why do we put public API to internal folder?
I would  except to see 
Public API for histogram in `.../dpl/pstl/glue_algorithm_def.h `
Host Pattern - in `.../dpl/pstl/algorithm_impl.h `
hetero Pattern  - in `.../dpl/pstl/hetero/algorithm_impl_hetero.h `